### ISSUE TITLE
Windows+L passthrough to Arch Linux on Hyper-V

### ISSUE THREAD

**Issue by eirikb** [neutrinolabs/xrdp#2085]
Trying to get Windows+L to move focus to right in i3.  
Steps I have done so far:

1. Set "Apply Windows key combinations" to "Only the remote computer" https://stackoverflow.com/a/30737423
2. Disable Win+L on host https://superuser.com/a/1096597

<kbd>Windows</kbd>+<kbd>H</kbd> works fine and I can focus window on the left, but <kbd>Windows</kbd>+<kbd>L</kbd> does not work.

Here is output from **xev** when pressing and releasing <kbd>Windows</kbd>+<kbd>H</kbd>:
> KeyPress event, serial 28, synthetic NO, window 0x1e00001,
    root 0x50f, subw 0x0, time 7393899, (644,923), root:(5128,941),
    state 0x0, keycode 115 (keysym 0xffeb, Super_L), same_screen YES,
    XLookupString gives 0 bytes:
    XmbLookupString gives 0 bytes:
    XFilterEvent returns: False

> KeyPress event, serial 28, synthetic NO, window 0x1e00001,
    root 0x50f, subw 0x0, time 7394315, (644,923), root:(5128,941),
    state 0x40, keycode 43 (keysym 0x68, h), same_screen YES,
    XLookupString gives 1 bytes: (68) "h"
    XmbLookupString gives 1 bytes: (68) "h"
    XFilterEvent returns: False

> KeyRelease event, serial 28, synthetic NO, window 0x1e00001,
    root 0x50f, subw 0x0, time 7394522, (644,923), root:(5128,941),
    state 0x40, keycode 43 (keysym 0x68, h), same_screen YES,
    XLookupString gives 1 bytes: (68) "h"
    XFilterEvent returns: False

> KeyRelease event, serial 28, synthetic NO, window 0x1e00001,
    root 0x50f, subw 0x0, time 7395099, (644,923), root:(5128,941),
    state 0x40, keycode 115 (keysym 0xffeb, Super_L), same_screen YES,
    XLookupString gives 0 bytes:
    XFilterEvent returns: False

Here is output from **xev** when pressing (real quick) <kbd>Windows</kbd>+<kbd>L</kbd>:
> KeyPress event, serial 28, synthetic NO, window 0x1e00001,
    root 0x50f, subw 0x0, time 7492466, (354,995), root:(4838,1013),
    state 0x0, keycode 115 (keysym 0xffeb, Super_L), same_screen YES,
    XLookupString gives 0 bytes:
    XmbLookupString gives 0 bytes:
    XFilterEvent returns: False

> KeyPress event, serial 28, synthetic NO, window 0x1e00001,
    root 0x50f, subw 0x0, time 7492866, (354,995), root:(4838,1013),
    state 0x40, keycode 50 (keysym 0xffe1, Shift_L), same_screen YES,
    XLookupString gives 0 bytes:
    XmbLookupString gives 0 bytes:
    XFilterEvent returns: False

> KeyRelease event, serial 28, synthetic NO, window 0x1e00001,
    root 0x50f, subw 0x0, time 7492866, (354,995), root:(4838,1013),
    state 0x41, keycode 50 (keysym 0xffe1, Shift_L), same_screen YES,
    XLookupString gives 0 bytes:
    XFilterEvent returns: False

> KeyRelease event, serial 28, synthetic NO, window 0x1e00001,
    root 0x50f, subw 0x0, time 7492867, (354,995), root:(4838,1013),
    state 0x40, keycode 115 (keysym 0xffeb, Super_L), same_screen YES,
    XLookupString gives 0 bytes:
    XFilterEvent returns: False

If I hold the keys down for more than say 1 second it will begin spamming this:
>KeyPress event, serial 28, synthetic NO, window 0x1e00001,
    root 0x50f, subw 0x0, time 7493368, (354,995), root:(4838,1013),
    state 0x0, keycode 46 (keysym 0x6c, l), same_screen YES,
    XLookupString gives 1 bytes: (6c) "l"
    XmbLookupString gives 1 bytes: (6c) "l"
    XFilterEvent returns: False

>KeyRelease event, serial 28, synthetic NO, window 0x1e00001,
    root 0x50f, subw 0x0, time 7493418, (354,995), root:(4838,1013),
    state 0x0, keycode 46 (keysym 0x6c, l), same_screen YES,
    XLookupString gives 1 bytes: (6c) "l"
    XFilterEvent returns: False

>KeyPress event, serial 28, synthetic NO, window 0x1e00001,
    root 0x50f, subw 0x0, time 7493418, (354,995), root:(4838,1013),
    state 0x0, keycode 46 (keysym 0x6c, l), same_screen YES,
    XLookupString gives 1 bytes: (6c) "l"
    XmbLookupString gives 1 bytes: (6c) "l"
    XFilterEvent returns: False

**Comment by eirikb** [comment]
I found a solution. A bit hacky, but it seems to work fine.
Idea is to remap left windows as another key I don't use as much, and use this key as mod in i3.

Hacked together like this:

In Windows:
1. Revert the disable Win+L from above.
2. Use sharpkeys (could probably use powertoys) to remap left windows key to right control key.

In Arch Linux:
1. Create a script as such:
```
xmodmap -e 'remove control = Control_R'
xmodmap -e 'add mod3 = Control_R'
````
2. Call the script as such: `sleep 2s && /home/eirikb/bah.sh` from `/etc/xrdp/startwm.sh` and `/etc/xrdp/reconnectwm.sh`.
3. In `~/.config/i3/config` I set `set $mod mod3`.

Notes: The sleep seems important, tried using .Xmodmap file but without success.
I guess the issue can be closed, but leaving it open in case someone got a comment, suggestions, or a fix for the actual problem.
